File: architecture-patterns/onion-architecture/java/src/com/example/domain/models/Category.java
package com.example.domain.models;

import lombok.Getter;

import java.util.UUID;

@Getter
public class Category {
    private final UUID id;
    private String name;
    private String description;

    public Category(UUID id, String name, String description) {
        validateName(name);
        
        this.id = id;
        this.name = name;
        this.description = description;
    }

    public static Category create(String name, String description) {
        return new Category(UUID.randomUUID(), name, description);
    }

    public void update(String name, String description) {
        validateName(name);
        
        this.name = name;
        this.description = description;
    }

    private void validateName(String name) {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("カテゴリ名は必須です");
        }
        if (name.length() > 50) {
            throw new IllegalArgumentException("カテゴリ名は50文字以内にしてください");
        }
    }
}

File: domain-driven-design/csharp/src/DDD.Domain/Products/Category.cs
namespace DDD.Domain.Products;

using DDD.Domain.Common;

public class Category : Entity<Guid>
{
    public string Name { get; private set; }
    public string Description { get; private set; }
    
    public Category(Guid id, string name, string description)
    {
        Id = id;
        Name = name;
        Description = description;
    }
    
    private Category() { } // For EF Core
    
    public void Update(string name, string description)
    {
        Name = name;
        Description = description;
    }
}